akka-actor-typed-tests/src/test/scala/akka/actor/typed/LogMessagesSpec.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import akka.actor
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.adapter._
+import akka.event.Logging
+import akka.testkit.EventFilter
+import org.scalatest.WordSpecLike
+
+class LogMessagesSpec extends ScalaTestWithActorTestKit("""
+    akka.loglevel = DEBUG # test verifies debug
+    akka.loggers = ["akka.testkit.TestEventListener"]
+    """) with WordSpecLike {
+
+  implicit val untyped: actor.ActorSystem = system.toUntyped
+
+  "The log messages behavior" should {
+    "log signals" in {
+      val behavior: Behavior[Signal] = Behaviors.logMessages(Behaviors.empty)
+
+      val ref: ActorRef[Signal] = spawn(behavior)
+
+      EventFilter.debug("received signal PostStop", source = ref.path.toString, occurrences = 1).intercept {
+        testKit.stop(ref)
+      }
+    }
+
+    "log messages" in {
+      val behavior: Behavior[String] = Behaviors.logMessages(Behaviors.empty)
+
+      val ref: ActorRef[String] = spawn(behavior)
+
+      EventFilter.debug("received message Hello", source = ref.path.toString, occurrences = 1).intercept {
+        ref ! "Hello"
+      }
+    }
+
+    "log messages with provided log level" in {
+      val opts = LogOptions().withLevel(Logging.InfoLevel)
+      val behavior: Behavior[String] = Behaviors.logMessages(opts, Behaviors.empty)
+
+      val ref: ActorRef[String] = spawn(behavior)
+
+      EventFilter.info("received message Hello", source = ref.path.toString, occurrences = 1).intercept {
+        ref ! "Hello"
+      }
+    }
+
+    "log messages with provided logger" in {
+      val logger = system.log
```

Comment:
I would have preferred to create some kind of mocked `Logger` here, but my understanding is that it would need to implement sending messages to the event stream in order for the `EventFilter` to pick it up.  Instead, I used the system log.  Is this acceptable?

Reply:
I think this is a perfectly fine way to check it 👍